packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -53,10 +65,33 @@ export function relation(definition?: Object) {
  * @param definition
  * @returns {(target:any, key:string)}
  */
-export function belongsTo(definition?: Object) {
-  // Apply model definition to the model class
-  const rel = Object.assign({type: RelationType.belongsTo}, definition);
-  return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
+export function belongsTo<T extends typeof Entity>(
+  targetModel: TypeResolver<T>,
+  definition?: Partial<BelongsToDefinition>,
+) {
+  const defIsCyclic =
+    definition &&
+    (definition as Object).hasOwnProperty('target') &&
+    !definition.target;
```

Comment:
We should refactor it to a utility function, such as `isTypeDefined(definition, key)`.